tests/shared/src/test/scala/scala/meta/tests/prettyprinters/SyntacticSuite.scala

```diff
@@ -1933,4 +1933,167 @@ class SyntacticSuite extends scala.meta.tests.parsers.ParseSuite {
     assertSyntax(tree, "\"\"\"\n\\\"\\\"\\\"\"\"\"")(tree)
   }
 
+  test("#2046 new line separator required before some Term.Block") {
+    assertEquals(
+      blockStat("{val a = 1; {b}; val c = 1}").syntax,
+      s"""
+      |{
+      |  val a = 1
+      |
+      |  {
+      |    b
+      |  }
+      |  val c = 1
+      |}
+      """.trim.stripMargin.replace("\n", EOL)
+    )
+    assertEquals(
+      blockStat("{val a = 1; {b}; {c}}").syntax,
+      s"""
+      |{
+      |  val a = 1
+      |
+      |  {
+      |    b
+      |  }
+      |  {
+      |    c
+      |  }
+      |}
+      """.trim.stripMargin.replace("\n", EOL)
+    )
+    assertEquals(
+      blockStat("{{a}; val b = 1; val c = 2; {d}}").syntax,
+      s"""
+      |{
+      |  {
+      |    a
+      |  }
+      |  val b = 1
+      |  val c = 2
+      |
+      |  {
+      |    d
+      |  }
+      |}
+      """.trim.stripMargin.replace("\n", EOL)
+    )
+
+    def testBlock(t: String, expectedStr: String, needSep: Boolean): Unit = {
+      def expected(sep: String): String =
+        s"""
+        |{
+        |${expectedStr.split("\n").map("  " + _).mkString("", "\n", sep)}
+        |  {
+        |    a
+        |  }
+        |}
+        """.trim.stripMargin.replace("\n", EOL)
```

Comment:
please put `.stripMargin` first, as `scalafmt` will not properly format this string otherwise (a bug in scalafmt, turns out).

Reply:
Ok, I "copied" this `.trim.stripMargin ...` from other tests in this file, should I also fix those?

Reply:
perhaps for others we don't need to fix it. i will try to fix the formatter later.